src/video.rs

```diff
@@ -0,0 +1,150 @@
+extern crate ggez;
+extern crate env_logger;
+
+use ggez::Context;
+use sdl2::video::{FullscreenType, DisplayMode};
+use log::LogLevel;
+
+#[derive(Debug, Clone)]
+struct DisplayModeManager {
+    index: usize,
+    modes: Vec<DisplayMode>,
+}
+
+impl DisplayModeManager {
+    pub fn new() -> DisplayModeManager {
+        DisplayModeManager {
+            index: 0,
+            modes: Vec::new(),
+        }
+    }
+
+    pub fn add_mode(&mut self, new_mode: DisplayMode) {
+            self.modes.push(new_mode);
+    }
+
+    pub fn print_supported_modes(&self) {
+        if log_enabled!(LogLevel::Info) {
```

Comment:
@Manghi It's not too late! Just push a fix to the `interface` branch in your repo. 👍